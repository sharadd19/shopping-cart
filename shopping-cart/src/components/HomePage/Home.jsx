import { Link } from "react-router-dom";
/* import HomeCarousel from "./HomeCarousel";
 */
export default function Home() {
  return (
    <>
      <h1 className="title">Welcome to fruits.</h1>
      <h3 className="summary">
        Discover our selection of fresh fruits, bursting with flavor and
        vitality. Delivered straight from the farm to your table.
      </h3>{
      <Link to="/store" >
        <button className="shop-now">Shop now</button>
      </Link>}
      {/* <HomeCarousel/> */}
    </>
  );
}
