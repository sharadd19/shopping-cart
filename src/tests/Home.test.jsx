import { render, screen } from "@testing-library/react";
import Home from "../components/HomePage/Home";
import { describe, it, expect } from "vitest";
import { MemoryRouter } from "react-router-dom";

describe("Home Component", () => {
  it("renders the home component and match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("shop now link should link to the store", () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const shopNowLink = screen.getByRole("link", { name: /shop now/i });
    expect(shopNowLink).toHaveAttribute("href", "/store");
  });

  it("contains the title", () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    const title = screen.getByRole("heading", { level: 1 });
    expect(title).toBeInTheDocument();
  });
});
