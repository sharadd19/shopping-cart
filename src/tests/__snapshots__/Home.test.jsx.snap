// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > renders the home component and match snapshot 1`] = `
<div>
  <div
    class="_home_8d9e9b"
  >
    <div>
      <h1
        class="_title_8d9e9b"
      >
        Welcome to SATO.
      </h1>
      <h3
        class="_summary_8d9e9b"
      >
        Discover our selection of electronics
      </h3>
      <a
        href="/store"
      >
        <button
          class="_shopButton_8d9e9b"
        >
          Shop now
        </button>
      </a>
    </div>
    <img
      alt=""
      src="./images/sankdisk.png"
    />
  </div>
</div>
`;
